src/prefect/deployments/runner.py

```diff
@@ -476,3 +612,188 @@ async def from_storage(
         cls._set_defaults_from_flow(deployment, flow)
 
         return deployment
+
+
+class DeploymentImage:
```

Comment:
This is a late addition. It felt better to have a class for Docker image config rather than overwhelming the `flow.deploy` and `deploy` signatures with Docker-related kwargs. I'm open to suggestions on the naming and implementation of this class.